### DISCUSSION TITLE
Py4JJavaError: An error occurred while calling z:com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline.

### DISCUSSION THREAD

**Discussion by naimisar** [JohnSnowLabs/spark-nlp#13069] [Q&A]
Hi,

I am getting following error while trying to load Pretrained Pipeline. Setup in Windows 10.

pipeline = PretrainedPipeline('explain_document_ml', lang='en')

**Error** : Py4JJavaError: An error occurred while calling z:com.johnsnowlabs.nlp.pretrained.PythonResourceDownloader.downloadPipeline.
: java.lang.UnsatisfiedLinkError: org.apache.hadoop.io.nativeio.NativeIO$POSIX.stat(Ljava/lang/String;)Lorg/apache/hadoop/io/nativeio/NativeIO$POSIX$Stat;

Spark NLP version 4.2.2
Apache Spark version: 3.3.1
openjdk version "1.8.0_352-352"
OpenJDK Runtime Environment (build 1.8.0_352-352-b08)
OpenJDK 64-Bit Server VM (build 25.352-b08, mixed mode)

Thanks & Regards,
Vishal Jani

**Comment by maziyarpanahi** [comment]
Spark, Hadoop, and TensorFlow on Windows are a bit tricky. If you follow all the steps here it should be fine: https://nlp.johnsnowlabs.com/docs/en/install#windows-support

In any case, Google Colab (CPU/GPU) can be also an alternative to a local machine on Windows: https://nlp.johnsnowlabs.com/docs/en/install#google-colab-notebook
